harness/tests/tensorboard/test_base.py

```diff
@@ -90,3 +93,63 @@ def test_illegal_type() -> None:
         tensorboard.build(
             env.det_cluster_id, env.det_experiment_id, env.det_trial_id, checkpoint_config
         )
+
+
+def test_upload_thread_normal_case() -> None:
+    upload_function = mock.Mock()
+    upload_thread = tensorboard.base._TensorboardUploadThread(upload_function)
+
+    upload_thread.start()
+    upload_thread.upload(
+        [pathlib.Path("test_value/file1.json"), pathlib.Path("test_value/file2.json")]
+    )
+    upload_thread.upload([pathlib.Path("test_value/file3.json")])
+    # Pass in sentinel value to exit thread
+    upload_thread.close()
+    upload_thread.join()
+
+    assert upload_function.call_count == 2
+
+
+def test_upload_thread_exception_case() -> None:
+    # 1. Define custom hook to capture threads fail
+    #    with exception. This hook is triggered when
+    #    a thread exits with exception.
+    #    Also store orignal excepthook for reset later.
+    import threading
+
+    threads_with_exception = set()
+
+    def custom_excepthook(args):
+        thread_name = args.thread.ident
+        threads_with_exception.add(thread_name)
+
+    original_excepthook = threading.excepthook
+
+    # 2. Define function that throws exception
+    def upload_function(paths: List[pathlib.Path]) -> None:
+        raise Exception("An exception is raised")
+
+    # 3. Set up a _TensorboardUploadThread instance
+    upload_thread = tensorboard.base._TensorboardUploadThread(upload_function)
+
+    try:
+        # 4. overwrite excepthook
```

Comment:
Last nit: you are over-commenting.  I really don't see how this comment adds anything beyond what the code already says.  Same for comments 2, 3, 5, 6, and 7.

I do like the overall comment in 1. though.

Reply:
I agree we can do without these comments. Will take note for future reference.